### ISSUE TITLE
TFX 1.2.0 Issues

### ISSUE THREAD

**Issue by dhruvesh09** [tensorflow/tfx#4159]
Please comment if you find any issues with TFX 1.2.0

Thanks

**Comment by ssoudan** [comment]
Hi, 

trying to get the Pusher to work with Vertex but seems that https://github.com/tensorflow/tfx/blob/13979ced62b50292a5dea7be2f15205fe8469400/tfx/extensions/google_cloud_ai_platform/pusher/executor.py#L135 is not propagating the enable_vertex flag and the execution fails. Did I miss something?

Thanks.

**Comment by hugoferrero** [comment]
Hi!... I'm trying to install tfx 1.2.0 locally (Ubuntu 20.04.1 LTS, python 3.8) and i get this "ERROR: Failed building wheel for google-cloud-profiler". I checked on "https://pypi.org/project/google-cloud-profiler/" and it seems that google-cloud-profiler package doesn't have any wheels........thanks in advance.

**Comment by jiyongjung0** [comment]
@hugoferrero Hi, thank you for reporting. It seems weird. Could you try install google-cloud-profiler alone? It is distributed as a source only package and it seems not related to TFX itself. You might be able to find hints from the [homepage](https://github.com/GoogleCloudPlatform/cloud-profiler-python).

**Comment by jiyongjung0** [comment]
@ssoudan Hi, thank you for the report. The flag is not needed to propagated. we are using `endpoint_region` to pass vertex specific configs. Could you give some more logs, codes or information about the failure?

**Comment by vaskozl** [comment]
I'm hitting a `TypeError: Object of type ResolverFunction is not JSON serializable` when compiling the pipelines with `kubeflow_dag_runner.KubeflowDagRunner`.

```
2021-08-25 10:35:00.733220: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
232021-08-25 10:35:00.733270: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
24WARNING:absl:`custom_executor_spec` is deprecated. Please customize component directly.
25Traceback (most recent call last):
26  File "dags/eb_lstm_pipeline_kubeflow.py", line 161, in <module>
27    kubeflow_dag_runner.KubeflowDagRunner(config=runner_config).run(tfx_pipeline)
28  File "/usr/local/lib/python3.8/dist-packages/tfx/orchestration/kubeflow/kubeflow_dag_runner.py", line 423, in run
29    self._parse_parameter_from_pipeline(pipeline)
30  File "/usr/local/lib/python3.8/dist-packages/tfx/orchestration/kubeflow/kubeflow_dag_runner.py", line 310, in _parse_parameter_from_pipeline
31    self._parse_parameter_from_component(component)
32  File "/usr/local/lib/python3.8/dist-packages/tfx/orchestration/kubeflow/kubeflow_dag_runner.py", line 284, in _parse_parameter_from_component
33    serialized_component = json_utils.dumps(component)
34  File "/usr/local/lib/python3.8/dist-packages/tfx/utils/json_utils.py", line 185, in dumps
35    return json.dumps(obj, cls=_DefaultEncoder, sort_keys=True)
36  File "/usr/lib/python3.8/json/__init__.py", line 234, in dumps
37    return cls(
38  File "/usr/local/lib/python3.8/dist-packages/tfx/utils/json_utils.py", line 95, in encode
39    return super().encode(obj)
40  File "/usr/lib/python3.8/json/encoder.py", line 199, in encode
41    chunks = self.iterencode(o, _one_shot=True)
42  File "/usr/lib/python3.8/json/encoder.py", line 257, in iterencode
43    return _iterencode(o, 0)
44  File "/usr/local/lib/python3.8/dist-packages/tfx/utils/json_utils.py", line 140, in default
45    return super().default(obj)
46  File "/usr/lib/python3.8/json/encoder.py", line 179, in default
47    raise TypeError(f'Object of type {o.__class__.__name__} '
48TypeError: Object of type ResolverFunction is not JSON serializable
```

**Comment by ssoudan** [comment]
Hi @jiyongjung0, it looks needed to me, also I have made the change in a fork and can ‘Push’ to Vertex while it is failing with 1.2.0 from upstream. 

Check the function here https://github.com/tensorflow/tfx/blob/13979ced62b50292a5dea7be2f15205fe8469400/tfx/extensions/google_cloud_ai_platform/runner.py#L253, passing `endpoint_region` is not enough.
More specifically: https://github.com/tensorflow/tfx/blob/13979ced62b50292a5dea7be2f15205fe8469400/tfx/extensions/google_cloud_ai_platform/runner.py#L317 .

**Comment by hugoferrero** [comment]
Hi @jiyongjung0. Thanks for response. I also tried to install google-cloud-profiler package alone and i got the same error. There is a comment in https://github.com/GoogleCloudPlatform/cloud-profiler-python/issues/98#issuecomment-905132655
abuot this issue with google-cloud-profiler.

**Comment by pilipenok** [comment]
@ssoudan 
I appreciate your help
but even with your updates I still have `InvalidArgument: 400 'automatic_resources' is not supported for Model ...`
seems it could be a problem with my Model and not with tfx

**Comment by sfines-clgx** [comment]
#3961 3961 Is also impacting me on 1.2.0

**Comment by deep-diver** [comment]
For those of you who just wants to use Vertex Uploader/Deployer, me and @sayakpaul had implemented custom components. In the mean time some of you could use those until the issue gets fixed.

**Comment by Arisiru** [comment]
Can not import TFX on macOS BigSur version 11.5.2
virtual environment created by `pipenv`

```
python --version
Python 3.8.2
```

```
import tensorflow as tf
import numpy as np
from tfx import v1 as tfx

print('TensorFlow version: %s' % (tf.__version__))
print('Numpy version: %s' % (np.__version__))
print('TFX version: %s' % (tfx.__version__))
```

Fails with 
```
Traceback (most recent call last):
  File "tfx.py", line 4, in <module>
    from tfx import v1 as tfx
  File "/Users/xxx/Projects/python/experiments/tfx-python-38/tfx.py", line 4, in <module>
    from tfx import v1 as tfx
ImportError: cannot import name 'v1' from partially initialized module 'tfx' (most likely due to a circular import) (/Users/xxxx/Projects/python/experiments/tfx-python-38/tfx.py)
```

pip show tfx
```
Name: tfx
Version: 1.2.0
Summary: TensorFlow Extended (TFX) is a TensorFlow-based general-purpose machine learning platform implemented at Google.
Home-page: https://www.tensorflow.org/tfx
Author: Google LLC
Author-email: tensorflow-extended-dev@googlegroups.com
License: Apache 2.0
Location: /Users/Alan_Spark/.local/share/virtualenvs/tfx-python-38-dj6H6SV9/lib/python3.8/site-packages
Requires: kubernetes, ml-pipelines-sdk, keras-tuner, pyarrow, tensorflow-data-validation, tensorflow, ml-metadata, google-cloud-bigquery, packaging, numpy, pyyaml, tfx-bsl, google-api-python-client, google-cloud-aiplatform, absl-py, google-apitools, tensorflow-serving-api, grpcio, docker, protobuf, click, portpicker, attrs, apache-beam, tensorflow-hub, jinja2, tensorflow-transform, tensorflow-model-analysis
Required-by: 
```

**Comment by jiyongjung0** [comment]
@Arisiru Thank you for the report. It seems weird...

Could you rename the failing python file other than `tfx.py` and retry? Python might confuse the file with the tfx package.

**Comment by Arisiru** [comment]
@jiyongjung0 
> Could you rename the failing python file other than `tfx.py` and retry? Python might confuse the file with the tfx package.

Amazing! It actually fix the problem. You save my day!

**Comment by pilipenok** [comment]
Hi @jiyongjung0,
Just dependency warning  between tfx==1.2.0 and google-cloud-aiplatform

When create a Vertex Pipeline Run via `kfp.v2.google.client.AIPlatformClient(...).create_run_from_job_spec(...)` like in penguin example
I see 
```FutureWarning: AIPlatformClient will be deprecated in v1.9. Please use PipelineJob https://googleapis.dev/python/aiplatform/latest/_modules/google/cloud/aiplatform/pipeline_jobs.html in Vertex SDK. Install the SDK using "pip install google-cloud-aiplatform"```

but 
`tfx 1.2.0 depends on google-cloud-aiplatform<0.8 and >=0.5.0` which doesn't support Vertex PipelineJob yet. (the last for now is 1.4.3 and it does)

can't wait for a tfx==1.3.0!
Thanks, Oleg

**Comment by jiyongjung0** [comment]
@pilipenok @ssoudan Sorry for late follow-up, but the Vertex pusher issue was fixed in the latest 1.3.1 release.

Example usage: (Thank you, @pilipenok for letting me know the possible pitfalls!)
```
  vertex_serving_spec = {
      'project_id': project_id,
      'endpoint_name': 'my_endpoint_20211008',
      'machine_type': 'n1-standard-2',
  }

  pusher = tfx.extensions.google_cloud_ai_platform.Pusher(
      model=trainer.outputs['model'],
      custom_config={
          tfx.extensions.google_cloud_ai_platform.ENABLE_VERTEX_KEY:
              True,
          tfx.extensions.google_cloud_ai_platform.VERTEX_REGION_KEY:
              'us-central1',
          tfx.extensions.google_cloud_ai_platform.VERTEX_CONTAINER_IMAGE_URI_KEY:
              'us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-6:latest',
          tfx.extensions.google_cloud_ai_platform.SERVING_ARGS_KEY:
            vertex_serving_spec,
      })
```

**Comment by venkat2469** [comment]
No Active issues found on TFX 1.2.0. will reopen if any issues found in future.
